Xplat/src/main/java/vazkii/botania/common/block/flower/functional/LooniumBlockEntity.java

```diff
@@ -197,48 +214,102 @@ private ItemStack pickRandomLoot(ServerLevel world, RandomSource rand) {
+			if (structureConfig != null && (structureEntry.getBooleanValue()
+					|| structureConfig.boundingBoxType == StructureSpawnOverride.BoundingBoxType.STRUCTURE)) {
+				structureConfigs.put(structureEntry.getKey(), structureConfig);
+			}
+		}
+
+		structureConfigs.put(LooniumStructureConfiguration.DEFAULT_CONFIG_ID, getDefaultConfig(configData));
+		return structureConfigs;
+	}
+
+	private static LooniumStructureConfiguration getDefaultConfig(ConfigDataManager configData) {
+		var defaultConfig = configData.getEffectiveLooniumStructureConfiguration(
+				LooniumStructureConfiguration.DEFAULT_CONFIG_ID);
+		return defaultConfig != null ? defaultConfig : FALLBACK_CONFIG.get();
+	}
+
+	private void detectStructure(ServerLevel world) {
+		// structure ID and whether the position is inside a structure piece (false = only overall bounding box)
+		var structureMap = new Object2BooleanRBTreeMap<ResourceLocation>();
 		StructureManager structureManager = world.structureManager();
 		BlockPos pos = getBlockPos();
 		var structures = structureManager.getAllStructuresAt(pos);
 		for (var entry : structures.entrySet()) {
 			Structure structure = entry.getKey();
 			var start = structureManager.getStructureAt(pos, structure);
 			if (start.isValid()) {
-				ResourceLocation structureId = world.registryAccess().registryOrThrow(Registries.STRUCTURE).getKey(structure);
+				ResourceLocation structureId =
+						world.registryAccess().registryOrThrow(Registries.STRUCTURE).getKey(structure);
 				boolean insidePiece = structureManager.structureHasPieceAt(pos, start);
-				BotaniaAPI.LOGGER.info("Found structure {}, inside piece: {}", structureId, insidePiece);
-				detectedStructures.add(ObjectBooleanPair.of(structureId, insidePiece));
+				if (insidePiece || !structureMap.getBoolean(structureId)) {
+					structureMap.put(structureId, insidePiece);
+				}
 			}
 		}
 
-		if (detectedStructures.isEmpty()) {
-			// not within any structures, keep default loot table
-			return;
-		}
-
-		var lootTableCandidates = new ArrayList<ResourceLocation>(detectedStructures.size());
-		for (var entry : detectedStructures) {
-			// TODO: grab structure configuration data from registry (assume must be inside a piece for now)
-			if (!entry.valueBoolean()) {
-				continue;
-			}
-			var structureId = entry.key();
-			var candidateId = prefix("loonium/%s/%s".formatted(structureId.getNamespace(), structureId.getPath()));
-			LootTable lootTable = world.getServer().getLootData().getLootTable(candidateId);
-			if (lootTable != LootTable.EMPTY) {
-				lootTableCandidates.add(candidateId);
-			}
+		if (structureMap.isEmpty()) {
+			detectedStructures = Object2BooleanMaps.emptyMap();
+		} else if (structureMap.size() == 1) {
+			var key = structureMap.firstKey();
+			detectedStructures = Object2BooleanMaps.singleton(key, structureMap.getBoolean(key));
+		} else {
+			detectedStructures = new Object2BooleanArrayMap<>(structureMap);
```

Comment:
Is this check necessary? Can't you just use the constructor like this in the empty and singleton cases too?